### PR TITLE
Bw view published posts

### PR DESCRIPTION
**Pull request by Jennifer-Blue-Williams**
#54 View All Published Posts

Added file Post.cs to Models directory.

Added file IPostRepository.cs and PostRepository.cs to Repository directory.

Added file PostController.cs to Controller Repository.

Added IPostRepository/PostRepository to Startup.cs.

Added file postManager.js to modules directory.

Added PostList.js to components directory.

Added a post route path to ApplicationViews.js
​
#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout bw-ViewPublishedPosts

    ```
2. Open a new Terminal tab (⌘T) and navigate to the server directory.
3. Test app functionality.

When user selects the Posts menu option
Then they should be directed to the Posts list page
showing the title, author and category. It should ONLY contain approved Posts
It should ONLY contain Posts with a publication date that is in the past
It should be in order of publication date with the most recent on top
 
4. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.

### CODE REVIEW

File: Tabloid/Controllers/PostController.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Tabloid.Repositories;
+using System.Collections.Generic;
+using Tabloid.Models;
+using System;
+
+namespace Tabloid.Controllers
+{
+    [Route("api/[controller]")]
```
Comment (aliparadise) [inline Tabloid/Controllers/PostController.cs:9]: May want to add [Authorize] here

File: Tabloid/Controllers/PostController.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Tabloid.Repositories;
+using System.Collections.Generic;
+using Tabloid.Models;
+using System;
+
+namespace Tabloid.Controllers
+{
```
Comment (aliparadise) [inline Tabloid/Controllers/PostController.cs:8]: May want to add [Authorize] here.

Comment (Jennifer-Blue-Williams) [reply Tabloid/Controllers/PostController.cs:7]: Added [Authorize] to the PostController per review notes.

File: Tabloid/client/src/modules/postManager.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+const baseUrl = "/api/post";
+
+export const GetAllPublishedPosts = () => {
+  return fetch(baseUrl).then((res) => res.json());
```
Comment (joshdbarton) [inline Tabloid/client/src/modules/postManager.js:4]: you're going to have to use the `getToken()` function here if you've added the `Authorize` attribute to the PostController in your API.

**Review by joshdbarton — CHANGES_REQUESTED** [review]

**Review by joshdbarton — APPROVED** [review]

File: Tabloid/client/src/modules/postManager.js (hunk 3 above)
Comment (Jennifer-Blue-Williams) [reply Tabloid/client/src/modules/postManager.js]: Thank you for pointing that out. All is resolved and the code is working correctly now.

**Pull request merged**